File: 3889-mirror-frequency-distance/3889-mirror-frequency-distance.cpp
class Solution {
public:
    int mirrorFrequency(string s) {
        int n = s.length();
        vector<int> digits(10, 0), chars(26, 0);
        int sum = 0;

        for (int i = 0; i < n; ++i) {
            if (s[i] >= '0' && s[i] <= '9')
                digits[s[i] - '0']++;
            else
                chars[s[i] - 'a']++;
        }

        for (int i = 0; i < 13; i++) {
            int fc = chars[i];
            int fm = chars[25 - i];

            sum += fabs(fc - fm);
        }

        for (int i = 0; i < 5; i++) {
            int fc = digits[i];
            int fm = digits[9 - i];

            sum += fabs(fc - fm);
        }
        return sum;
    }
};

File: 3889-mirror-frequency-distance/3889-mirror-frequency-distance.py
class Solution:
    def mirrorFrequency(self, s: str) -> int:
        chars = [0] * 26
        digits = [0] * 10
        for ch in s:
            if ch >= "0" and ch <= "9":
                digits[ord(ch) - 48] += 1
            else:
                chars[ord(ch) - 97] += 1

        Sum = 0
        for i in range((26 >> 1)):
            fc, fm = chars[i], chars[25 - i]
            Sum += abs(fc - fm)

        for i in range((10 >> 1)):
            fc, fm = digits[i], digits[9 - i]
            Sum += abs(fc - fm)

        return Sum
